GoogleCloudPlatform/google-cloud-dotnet
# Move from project.json to csproj

- Remove all project.json and xproj files
- Create a new tool to generate project files for:
  - Almost all packages we publish (exception: Google.Logging.Type)
  - Snippet, test and integration test projects
  The tool reads apis/apis.json. We could decide to split the files
  out, one per API, if we wanted - this makes it very easy to change
  versions in bulk though, and compare projects.
- Update all test projects to net452 to work with dotnet test + xunit
- Manually migrate other projects
- Move the Log4Net package out of Google.Logging.V2, and give it
  its own set of documentation
- Rename all Google.Cloud.Language.V1.Experimental directories to
  include Experimental
- Fix documentation generation:
  - Project layout changes
  - Parameter handling, which appears to have been broken for a long
    time, but we weren't generating all the snippets we should have
    been (!)
- Delete UpdateVersions tool
- Update all build scripts
  - We now don't run coverage or pack the projects in build.sh
  - build.sh auto-creates a giant sln file to restore and build
  - Individual test csproj files are then run
- Fixed enum test that had duplicate inputs (causing xunit to die)

Commits:

- Manual changes
- Moving Google.Logging.Log4Net
- Moving Google.Language.V1.Experimental
- All the global.json, project.json, xproj, sln and csproj files
